onmt/inference_engine_ct2.py

```diff
@@ -0,0 +1,135 @@
+import ctranslate2
+import pyonmttok
+from onmt.constants import CorpusTask, DefaultTokens, ModelTask
+from onmt.inputters.dynamic_iterator import build_dynamic_dataset_iter
+from onmt.inputters.inputter import IterOnDevice
+from onmt.transforms import get_transforms_cls
+from onmt.utils.logging import logger
+
+
+class InferenceEngineCT2(object):
+
+    """Wrapper Class to run Inference with ctranslate2.
+
+    Args:
+        opt: inference options
+    """
+
+    def __init__(self, opt):
+        self.opt = opt
+        self.transforms_cls = get_transforms_cls(self.opt._all_transform)
+        self.logger = logger
+        if opt.world_size == 1:
+            self.device_id = 0
+
+    def warm_up(self):
+        self.translator = self.build_ct2_translator(self.opt)
+        self.build_tokenizer()
+
+    def build_ct2_translator(self, opt):
+        # add model_task in config_file and opts.model_opts(parser) in  _get_parser
+        if opt.model_task == ModelTask.LANGUAGE_MODEL:
+            translator = ctranslate2.Generator(
+                opt.models[0], device="cuda", device_index=opt.gpu_ranks
+            )
+        else:
+            translator = ctranslate2.Translator(
+                self.opt.models[0], device="cuda", device_index=opt.gpu_ranks
+            )
+        return translator
+
```

Comment:
not sure why _translate() and translate_batch()
is one of the two supposed to be exposed outside ?
if not then use _translate_batch()